src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+        }
+    }
+
+    // ArrayList for storing user-added tasks.
+    private static final ArrayList<Task> taskList = new ArrayList<>();
+
+    // Greets the user
+    private static void greet() {
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\t Greetings! I'm Barry.\n" + "\t How could I assist you?");
+        System.out.println("\t [Type help for a list of available commands]");
+        System.out.println("\t____________________________________________________________\n");
+    }
+
+    // Exits the program with a message
+    private static void exit() {
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\t Bye, see you next time! :)");
+        System.out.println("\t____________________________________________________________\n");
+    }
+
+    // List all available commands
+    private static void help() {
+        System.out.println("\t____________________________________________________________");
+        System.out.println("\t Here's a list of available commands:");
+        System.out.println("\t   help                                           List all available commands");
+        System.out.println("\t   bye                                            Quit");
+        System.out.println("\t   list                                           List all tasks in your list");
+        System.out.println("\t   mark [x]                                       Mark task number x in your list");
+        System.out.println("\t   unmark [x]                                     Unmark task number x in your list");
+        System.out.println("\t   delete [x]                                     Delete task number x in your list");
+        System.out.println("\t   todo [task name]                               Add a ToDo to your list");
+        System.out.println("\t   deadline [task name] /by [deadline]            Add a Deadline to your list");
+        System.out.println("\t   event [task name] /from [start] /to [end]      Add an Event to your list");
+        System.out.println("\t____________________________________________________________\n");
+    }
+
+    // Task list mode, allows user to enter tasks into a list, and print all tasks in the list
+    // currently with "list", exits with "bye"
+    private static void taskListMode() {
+        Scanner scanner = new Scanner(System.in);
+        final Pattern markPattern = Pattern.compile("^mark\\s(\\d+)$");
+        final Pattern unmarkPattern = Pattern.compile("^unmark\\s(\\d+)$");
+        final Pattern deletePattern = Pattern.compile("^delete\\s(\\d+)$");
+        final Pattern todoPattern = Pattern.compile("^todo\\s(.+)$");
+        final Pattern deadlinePattern = Pattern.compile("^deadline\\s(.+)\\s/by\\s(.+)$");
+        final Pattern eventPattern = Pattern.compile("^event\\s(.+)\\s/from\\s(.+)\\s/to\\s(.+)$");
```

Comment:
LGTM. I like the idea of this and how it uses regex to capture the different patterns.